Ol operator update (#48)
OpenLiberty/guide-openliberty-operator-intro: README.adoc

// Static guide instruction
ifndef::cloud-hosted[]
== Additional prerequisites

> You need a Linux machine to run this guide. Before you begin, your Linux need a containerization software for building containers. You will use `Docker` in this guide. For Docker installation instructions, refer to the official https://docs.docker.com/install/[Docker documentation^].

Reviewer: ```suggestion
You must run this guide in a Linux environment. You will also use Docker. For Docker installation instructions, see the official https://docs.docker.com/install/[Docker documentation^].
```

> You will use `Minikube` as a single-node {kube} cluster that runs locally in your Linux. Make sure you have `kubectl` installed. If you need to install `kubectl`, see the https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl-on-linux[kubectl installation instructions^]. For Minikube installation instructions, see the https://github.com/kubernetes/minikube#installation[Minikube documentation^].

Reviewer: ```suggestion
You will use `Minikube` as a single-node {kube} cluster that runs locally in your Linux environment. Make sure you have `kubectl` installed. If you need to install `kubectl`, see the https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl-on-linux[kubectl installation instructions^]. For Minikube installation instructions, see the https://github.com/kubernetes/minikube#installation[Minikube documentation^].
```

// =================================================================================================
// Static guide instruction
ifndef::cloud-hosted[]
== Starting and preparing your cluster for deployment

Start your {kube} cluster.

Run the following command from a command-line session:

[role=command]
```
minikube start
```

> If you run Minikube by root user, you can append the `--force` option to the above command.

Reviewer: ```suggestion
If you run Minikube as a root user, you can append the `--force` option to the previous command.
```

> Next, validate that you have a healthy {kube} environment by running the following command from the active command-line session.

Reviewer: ```suggestion
Next, validate that you have a healthy `{kube}` environment by running the following command from the active command-line session.
```

Reviewer 2: {kube} will convert to Kubernetes, i.e. "...a healthy Kubernetes environment..."
 
 if add \`s, it will be   "...a healthy `Kubernetes` environment..."

[role='command']
```
kubectl get nodes
```

> This command should return a `Ready` status for the `minikube` node.

Reviewer: ```suggestion
If your environment is healthy, this command returns a `Ready` status for the `minikube` node.
```

[role='no_copy']
```
NAME       STATUS   ROLES           AGE     VERSION
minikube   Ready    control-plane   3h48m   v1.26.1
```

> Run the following command to configure the Docker CLI to use Minikube's Docker daemon. After you run this command, you will be able to interact with Minikube's Docker daemon and build new images directly to it from your host machine:

Reviewer: ```suggestion
Run the following command to configure the Docker CLI to use Minikube's Docker daemon. After you run this command, you can interact with Minikube's Docker daemon and build new images directly to it from your host machine:
```

```
eval $(minikube docker-env)
```
endif::[]

== Installing the Operator

// Static guide instruction
ifndef::cloud-hosted[]

> Before you can deploy your microservice, you must install the https://cert-manager.io[cert-manager^] and the Open Liberty Operator. You should read the official installation document on the link:https://github.com/OpenLiberty/open-liberty-operator/tree/main/deploy/releases/1.2.0#readme[GitHub repository].

Reviewer: ```suggestion
Before you can deploy your microservice, you must install the https://cert-manager.io[cert-manager^] and the Open Liberty Operator. For more information, see the link:https://github.com/OpenLiberty/open-liberty-operator/tree/main/deploy/releases/1.2.0#readme[installation instructions].
```

[role='command']
```
kubectl apply --server-side -f https://raw.githubusercontent.com/OpenLiberty/open-liberty-operator/main/deploy/releases/1.2.0/kubectl/openliberty-app-crd.yaml
```
Custom Resources extend the Kubernetes API and enhance its functionality.

Set environment variables for namespaces for the Operator by running the following commands:

[role='command']
```
OPERATOR_NAMESPACE=default
WATCH_NAMESPACE='""'
```

Next, run the following commands to install cluster-level role-based access:

[role='command']
```
curl -L https://raw.githubusercontent.com/OpenLiberty/open-liberty-operator/main/deploy/releases/1.2.0/kubectl/openliberty-app-rbac-watch-all.yaml \
  | sed -e "s/OPEN_LIBERTY_OPERATOR_NAMESPACE/${OPERATOR_NAMESPACE}/" \
  | kubectl apply -f -
```

Finally, run the following commands to install the Operator:

[role='command']
```
curl -L https://raw.githubusercontent.com/OpenLiberty/open-liberty-operator/main/deploy/releases/1.2.0/kubectl/openliberty-app-operator.yaml \
  | sed -e "s/OPEN_LIBERTY_WATCH_NAMESPACE/${WATCH_NAMESPACE}/" \
  | kubectl apply -n ${OPERATOR_NAMESPACE} -f -
```
endif::[]

// Cloud hosted guide instruction
ifdef::cloud-hosted[]

> In this Skills Network environment, the Open Liberty Operator is already installed by the administrator. If you like to learn how to install the Open Liberty Operator, you can learn from the [Deploying microservices to OpenShift by using Kubernetes Operators](https://openliberty.io/guides/cloud-openshift-operator.html#installing-the-operators) guide or the Open Liberty Operator [document](https://github.com/OpenLiberty/open-liberty-operator/tree/main/deploy/releases/1.2.0#readme).

Reviewer: ```suggestion
The Open Liberty Operator is already installed in this Skills Network environment. To learn how to install the Open Liberty Operator yourself, see the [Deploying microservices to OpenShift by using Kubernetes Operators](https://openliberty.io/guides/cloud-openshift-operator.html#installing-the-operators) guide or the [Open Liberty Operator documentation](https://github.com/OpenLiberty/open-liberty-operator/tree/main/deploy/releases/1.2.0#readme).
```
